src/Tizen.NUI.Components/Controls/Button.cs

```diff
@@ -277,55 +279,133 @@ public enum IconOrientation
             get
             {
-                return Style?.Text?.Text?.GetValue(ControlState);
+                if (null == overlayImage)
+                {
+                    overlayImage = CreateOverlayImage();
+                    if (null != Extension)
+                    {
+                        overlayImage = Extension.OnCreateOverlayImage(this, overlayImage);
+                    }
+                    overlayImage.WidthResizePolicy = ResizePolicyType.FillToParent;
+                    overlayImage.HeightResizePolicy = ResizePolicyType.FillToParent;
+                    
+                    Add(overlayImage);
+                }
+                return overlayImage;
             }
-            set
+            internal set
             {
-                if (null != Style?.Text)
+                overlayImage = value;
+            }
+        }
+
+        /// <summary>
+        /// Button's text part.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public TextLabel ButtonText
+        {
+            get
+            {
+                if (null == buttonText)
                 {
-                    Style.Text.Text = value;
+                    buttonText = CreateText();
+                    if (null != Extension)
+                    {
+                        buttonText = Extension.OnCreateText(this, buttonText);
+                    }
+                    buttonText.HorizontalAlignment = HorizontalAlignment.Center;
+                    buttonText.VerticalAlignment = VerticalAlignment.Center;
+                    
+                    Add(buttonText);
                 }
+                return buttonText;
+            }
+            internal set
+            {
+                buttonText = value;
             }
         }
 
         /// <summary>
-        /// Flag to decide Button can be selected or not.
+        /// Return a copied Style instance of Button
+        /// </summary>
+        /// <remarks>
+        /// It returns copied Style instance and changing it does not effect to the Button.
+        /// Style setting is possible by using constructor or the function of ApplyStyle(ViewStyle viewStyle)
+        /// </remarks>
+        /// <since_tizen> 8 </since_tizen>
+        //public new ButtonStyle Style
+        //{
+        //    get
+        //    {
+        //        return new ButtonStyle(ViewStyle as ButtonStyle);
+        //    }
+        //}
+        public new ButtonStyle Style => ViewStyle as ButtonStyle;
+
+        /// <summary>
+        /// The text of Button.
         /// </summary>
         /// <since_tizen> 6 </since_tizen>
-        public bool IsSelectable
+        public string Text
         {
             get
             {
-                return (bool)GetValue(IsSelectableProperty);
+                return ButtonText.Text;
             }
             set
             {
-                SetValue(IsSelectableProperty, value);
+                ButtonText.Text = value;
```

Comment:
I think this patch (a) still has binding with corresponding Style, so following code should work.

```
Button button = new Button();
button.Text = "ABC";
Tizen.Log.Info("NUI", button.Style.Text.Text); //-> ABC
```

Reply:
All properties in components should work same as before.

Reply:
OK. Some codes need to  revert